packages/application-shell-connectors/src/utils/get-mc-api-url/get-mc-api-url.spec.ts

```diff
@@ -0,0 +1,32 @@
+import getMcApiUrl from './get-mc-api-url';
+
+describe.each`
+  servedByProxy | origin                                                                                         | mcApiUrlFromConfig                                     | expectedMcApiUrl
+  ${'false'}    | ${'http://localhost:3001'}                                                                     | ${'https://mc-api.europe-west1.gcp.commercetools.com'} | ${'https://mc-api.europe-west1.gcp.commercetools.com'}
+  ${'false'}    | ${'https://mc.commercetools.com'}                                                              | ${'https://mc-api.commercetools.com'}                  | ${'https://mc-api.commercetools.com'}
+  ${'false'}    | ${'https://mc.europe-west1.gcp.commercetools.com'}                                             | ${'https://mc-api.europe-west1.gcp.commercetools.com'} | ${'https://mc-api.europe-west1.gcp.commercetools.com'}
+  ${'true'}     | ${'https://mc.europe-west1.gcp.commercetools.com'}                                             | ${'https://mc-api.europe-west1.gcp.commercetools.com'} | ${'https://mc-api.europe-west1.gcp.commercetools.com'}
+  ${'true'}     | ${'https://mc.europe-west1.gcp.commercetools.com'}                                             | ${'https://mc-api.us-central1.gcp.commercetools.com'}  | ${'https://mc-api.europe-west1.gcp.commercetools.com'}
+  ${'true'}     | ${'https://mc-12345.europe-west1.gcp.commercetools.com'}                                       | ${'https://mc-api.europe-west1.gcp.commercetools.com'} | ${'https://mc-api.europe-west1.gcp.commercetools.com'}
+  ${'true'}     | ${'https://mc-12345.us-central1.gcp.commercetools.com'}                                        | ${'https://mc-api.europe-west1.gcp.commercetools.com'} | ${'https://mc-api.us-central1.gcp.commercetools.com'}
+  ${'true'}     | ${'https://123.mc-preview.europe-west1.gcp.commercetools.com'}                                 | ${'https://mc-api.europe-west1.gcp.commercetools.com'} | ${'https://mc-api.europe-west1.gcp.commercetools.com'}
+  ${'true'}     | ${'https://123.mc-preview.us-central1.gcp.commercetools.com'}                                  | ${'https://mc-api.europe-west1.gcp.commercetools.com'} | ${'https://mc-api.us-central1.gcp.commercetools.com'}
+  ${'true'}     | ${'https://test.mc-preview.us-central1.gcp.commercetools.com'}                                 | ${'https://mc-api.europe-west1.gcp.commercetools.com'} | ${'https://mc-api.us-central1.gcp.commercetools.com'}
+  ${'true'}     | ${'https://this-is-not-what-you-are-looking-for.mc-preview.us-central1.gcp.commercetools.com'} | ${'https://mc-api.europe-west1.gcp.commercetools.com'} | ${'https://mc-api.us-central1.gcp.commercetools.com'}
```

Comment:
The tests have been rewritten using `describe.each` and should cover more cases:

<img width="994" alt="image" src="https://user-images.githubusercontent.com/1110551/180470440-b0861d25-4614-49b5-a07d-9e4e1d394f4c.png">